pebblo/app/service/discovery_service.py

```diff
@@ -145,11 +151,32 @@ def process_request(self):
             )
             self._write_file_content_to_path(ai_apps.dict(), file_path)
 
-            logger.debug("AiApp discovery request completed successfully")
-            return {"message": "App Discover Request Processed Successfully"}
+            docs = ai_apps.dict()
+            doc_obj = DiscoverAIApps(
+                name=docs["name"],
+                description=docs["description"],
+                owner=docs["owner"],
+                instanceDetails=docs["instanceDetails"],
+                framework=docs["framework"],
+                lastUsed=docs["lastUsed"],
+                pebbloServerVersion=docs["pebbloServerVersion"],
+                pebbloClientVersion=docs["pebbloClientVersion"],
+            )
+            message = "App Discover Request Processed Successfully"
+            logger.debug(message)
+            response = DiscoverAIAppsResponseModel(docs=doc_obj, message=message)
+            return PebbloJsonResponse.build(
+                body=response.dict(exclude_none=True), status_code=200
+            )
         except ValidationError as ex:
-            logger.error(f"Error in process_request. Error:{ex}")
-            raise HTTPException(status_code=400, detail=str(ex))
+            response = DiscoverAIAppsResponseModel(docs=None, message=str(ex))
+            logger.error(f"Error in Discovery API process_request. Error:{ex}")
+            return PebbloJsonResponse.build(
+                body=response.dict(exclude_none=True), status_code=400
+            )
         except Exception as ex:
-            logger.error(f"Error in process_request. Error:{ex}")
-            raise HTTPException(status_code=500, detail=str(ex))
+            response = DiscoverAIAppsResponseModel(docs=None, message=str(ex))
+            logger.error(f"Error in Discovery API process_request. Error:{ex}")
+            return PebbloJsonResponse.build(
+                body=response.dict(exclude_none=True), status_code=500
```

Comment:
Till now, this API call was async, now it would be synchronous. Please test it with 500 response, so that we make sure it is being consumed as expected.

Reply:
Tested, working as expected